docs/querying/sql.md

```diff
@@ -115,14 +115,19 @@ milliseconds since 1970-01-01 00:00:00 UTC, not counting leap seconds. Therefore
 timezone information, but only carry information about the exact moment in time they represent. See the
 [Time functions](#time-functions) section for more information about timestamp handling.
 
-Druid generally treats NULLs and empty strings interchangeably, rather than according to the SQL standard. As such,
-Druid SQL only has partial support for NULLs. For example, the expressions `col IS NULL` and `col = ''` are equivalent,
+### Default Null handling
+By default Druid treats NULLs and empty strings interchangeably, rather than according to the SQL standard. As such,
+in this mode Druid SQL only has partial support for NULLs. For example, the expressions `col IS NULL` and `col = ''` are equivalent,
 and both will evaluate to true if `col` contains an empty string. Similarly, the expression `COALESCE(col1, col2)` will
 return `col2` if `col1` is an empty string. While the `COUNT(*)` aggregator counts all rows, the `COUNT(expr)`
 aggregator will count the number of rows where expr is neither null nor the empty string. String columns in Druid are
 NULLable. Numeric columns are NOT NULL; if you query a numeric column that is not present in all segments of your Druid
 datasource, then it will be treated as zero for rows from those segments.
 
+### SQL compatible null handling
```

Comment:
The header is weird here. It includes a bunch of stuff that isn't null handling related. I think it would be better to merge this stuff into the prior paragraph about null handling. I think we should also point out that the SQL optimizer works best when SQL compliant null handling is enabled, so if you're doing a lot of SQL with NULLs, it's best to enable the mode.

Reply:
combined sections and mentioned optimizer